codec/codec-fury/src/main/java/com/alipay/sofa/rpc/codec/fury/FurySerializer.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.alipay.sofa.rpc.codec.fury;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import com.alipay.sofa.rpc.codec.AbstractSerializer;
+import com.alipay.sofa.rpc.common.RemotingConstants;
+import com.alipay.sofa.rpc.common.struct.ConcurrentHashSet;
+import com.alipay.sofa.rpc.common.utils.CodecUtils;
+import com.alipay.sofa.rpc.common.utils.StringUtils;
+import com.alipay.sofa.rpc.config.ConfigUniqueNameGenerator;
+import com.alipay.sofa.rpc.context.RpcInvokeContext;
+import com.alipay.sofa.rpc.core.exception.SofaRpcException;
+import com.alipay.sofa.rpc.core.request.SofaRequest;
+import com.alipay.sofa.rpc.core.response.SofaResponse;
+import com.alipay.sofa.rpc.ext.Extension;
+import com.alipay.sofa.rpc.transport.AbstractByteBuf;
+import com.alipay.sofa.rpc.transport.ByteArrayWrapperByteBuf;
+
+import io.fury.Fury;
+import io.fury.Language;
+import io.fury.serializer.CompatibleMode;
+
+/**
+ * @author lipan
+ */
+@Extension(value = "fury", code = 20)
+public class FurySerializer extends AbstractSerializer {
+
+    private final FuryHelper    furyHelper  = new FuryHelper();
+    private final Set<Class<?>> registerSet = new ConcurrentHashSet<>();
+    private final Fury          fury;
+
+    public FurySerializer() {
+        fury = Fury.builder()
+            .withLanguage(Language.JAVA)
+            .withRefTracking(true)
+            .withCodegen(true)
+            .withNumberCompressed(true)
+            .withCompatibleMode(CompatibleMode.COMPATIBLE)
+            .requireClassRegistration(false)
```

Comment:
thanks,i will to refer,give me little time

Reply:
And also remember to take this into consideration:

Since ClassRegistration is used to limit the class that can be serialized and deserialized, how will users use this mechanism.

if ClassRegistration is set false, do we need to do the registration?
if ClassRegistration is set true (which means registration should be under control), is it appropriate to do the registration each time serializing or to provide any mechanism to let user register or config.

Reply:
thansk,u are right ! i will try to refer.
 i think ,should give users a choice.
maybe fury  documentation will explain it. Let me see

Reply:
thanks,but i think,registration is required.
I see the document
> Do not disable class registration unless you can ensure your environment is indeed secure.

Reply:
Yup, but take this two into consideration: 
1. is it that "setting registration true and register when serializing/deserializing" same to "setting registration false"
2. wat if im sure i deploy sofa rpc framework under trusted environment? (u should provide choices for ur users)

Reply:
> Yup,in fact, even if it is not set to true, it can serialize normally in most cases, I think you can have two constructors, the default is false, but serialization problems, the user chooses the other one

Absolutely, we r also discussing which to use as default. But for here, u hv to build one for rpc users, and the implements can be controlled by configs, as which I mentioned at above.

Reply:
thanks , i get  it